src/modules/add-site/hooks/use-stepper.ts

```diff
@@ -18,32 +19,33 @@ interface StepperConfig {
 }
 
 interface StepperContext {
+	flow?: 'blueprint' | 'backup' | 'create';
 	steps: StepperStep[];
-	isVisible: boolean;
+	isVisible?: boolean;
 	actionButton?: {
 		label: string;
 		isVisible: boolean;
 	};
-	onSubmit: () => void;
-	canSubmit: boolean;
+	onSubmit?: () => void;
+	canSubmit?: boolean;
 }
 
 export function useStepper( config?: StepperConfig ): StepperContext {
 	const { __ } = useI18n();
 	const { location } = useNavigator();
 
-	const stepperConfig = useMemo( () => {
-		const blueprintSteps = [
+	const stepperConfig = useMemo( (): StepperContext | null => {
```

Comment:
`stepperConfig` always contains `flow` and `steps`. We can use Pick to extract those from the context type:

```suggestion
	const stepperConfig = useMemo( (): Pick< StepperContext, 'flow' | 'steps' > | null => {
```

Reply:
Actually, I should fix this constant name, too, as it should be `stepperContext` here and not `stepperConfig`. The config is passed to this hook, and context is returned.

I untangled the types of context and hook return value - using the same type for both caused those needs to use a partial type.